docs/source/guide/zne-5-theory.myst

```diff
@@ -0,0 +1,110 @@
+the gates are to be mapped as discussed previously, the process is known as *global
+folding*, while if only some gates in the circuit are to be folded, the procedure implemented
+is known as *local folding*. Thus, in either case noise caused due to physical implementations
+of unitary gates on gate-model devices is scaled by inserting additional gates
+in the circuit.
+
+```{figure} ../img/zne_global_folding.png
+---
+width: 500
+name: figzne_global
+---
+The diagram demonstrates how gates are inserted in a circuit for ZNE when the goal is to scale noise by folding the entire circuit (*global folding*).
+```
+
+```{figure} ../img/zne_local_folding.png
+---
+width: 500
+name: figzne_local
+---
+The diagram demonstrates how gates are inserted in a circuit for ZNE when the goal is to fold some gates in the circuit to scale noise (*local folding*).
+```
+
+The same idea applies to devices with pulse-level access for ZNE {cite}`Kandala_2019_Nature, Kandala_2019_Nature`.
+The noise of the device can be altered by increasing the time over which pulses
+are implemented but keeps the original circuit unchanged.
+
+```{figure} ../img/zne_pulse_stretching.png
+---
+width: 500
+name: figzne_pulse
+---
+The diagram demonstrates how pulse stretching is used to increase noise in a physical device for ZNE.
+```
+
+In both gate-model and pulse-model scenarios, let $\tau$ be a parameter quantifying
+noise in the circuit. When the scale factor $\lambda$ is used to increase the circuit noise,
+the noise parameter will also change dependent on $\lambda$ as $\tau^\prime = \lambda \tau$.
+It can be observed that by default, for $\lambda = 1$, the input circuit remains unchanged
+as the new noise parameter is the same as the initial noise parameter $\tau^\prime = \tau$.
+Note that here the initial noise parameter $\tau$ is the noise of the physical device without
+any scaling.
+
+Let $E(\tau)$ be some circuit observable described as a function of the noise parameter
+and $\rho$ be the state prepared by the quantum circuit of interest. The expectation
+value can be described as a function of noise parameter as following:
+
+$$
+\langle E(\tau^\prime) \rangle = \text{Tr}[\rho E(\tau^\prime)]
+$$
+
+We wish to estimate $\langle E(\tau=0) \rangle$ by calculating a range of different
+$\langle E(\tau^\prime) \rangle$ as dictated by different values of $\lambda$. In
+order to be consistent with the [notation introduced previously in the documentation](https://mitiq.readthedocs.io/en/latest/guide/pec-5-theory.html#error-cancellation)
+$\langle E(\tau=0) \rangle$ could be re-labeled as $\langle E\rangle_{\rm ideal}$ and the noisy expectation
+values $\langle E(\tau^\prime) \rangle$ could be re-labeled as
+$\langle E(\tau) \rangle_{\rm noisy}$.
+
+The ideal expectation value can be extrapolated as shown below, 
+
+$$
+\langle E\rangle_{\rm ideal} = \sum_{\lambda \in S} n_{\lambda} \langle E_{\lambda} \rangle_{\rm noisy} + \Delta
+$$
```

Comment:
I kept the original formulation of @purva but fixed two aspects:

- Directly define $E(\lambda)$ instead of $E(\tau)$ to avoid changing the notation in the middle of the discussion.
- Define the extrapolation as a best-fitting process. You used an extrapolation formula which is a linear combination of expectation values. This is my fault since I suggested this formula in a previous discussion but now I realized that it is not the most general case. For example, for exponential extrapolation, the linear formula doesn't apply.  

@purva, please have a look if this change  is clear and makes sense. If not, feel free to edit.

```suggestion
## Step 2: Extrapolate to the noiseless limit

In both gate-model and pulse-model scenarios, let $\tau$ be a parameter quantifying
noise level in the circuit and let $\tau^\prime = \lambda \tau$ the scaled noise level.

For $\lambda = 1$, the input circuit remains unchanged
since the noise level $\tau^\prime = \tau$ is the same as the noise of the physical device without
any scaling.

Let $\rho(\tau')$ be the state prepared by a noise scaled quantum circuit. The expectation
value of an observable $A$ can be described as a function of the noise scaling parameter as follows:

$$
\langle E(\lambda) \rangle = \text{Tr}[\rho(\tau') A] = \text{Tr}[\rho(\lambda \tau) A]
$$

The idea of ZNE is that one can estimate the ideal expectation value $\langle E(\lambda=0) \rangle$, by measuring a range of different
expectation values $\langle E(\lambda) \rangle$ for different values of $\lambda \ge 1$ and extrapolating
to the zero-noise limit.

In practice the extrapolation can be done as follows:
  1) Assume $E(\lambda)\simeq f(\lambda; p_1, p_2, ... p_m)$, where $f$ is an *extrapolation model*, i.e., a function
  of $\lambda$ depending on a set of real parameters $p_1, p_2, \dots, p_m$.
  2) Fit the function $f$ to the measured noise-scaled expectation values, obtaining an optimal set of 
  parameters $\tilde p_1, \tilde p_2, \dots \tilde p_m$.
  3) Evaluate the corresponding zero-noise limit, i.e., $f(0; \tilde p_1, \tilde p_2, \dots \tilde p_m)$.

Different choices of $f$, produce different extrapolations. Typical choices for $f$ are: a linear function, a polynomial, an exponential.
For example, Richardson extrapolation, corresponds to the following polynomial model:


$$
f(\lambda; p_1, p_2, ... p_m) = p_1 + p_2 \lambda + p_3 \lambda^2 + \dots p_m \lambda^{m-1},
$$

where $m$ is equal to the number of data points in the fit (i.e. the number of noise scaled expectation values).
```